packages/devtools-core/tests/actions/index.test.ts

```diff
@@ -0,0 +1,36 @@
+import { model } from "../../src/model";
+import { initState } from "../../src/store";
+import { Action } from "../../src/types";
+import { recordAction, recordState } from "../../src/utils/communication";
+
+import "@babel/polyfill";
+import { testActionLog, testAppState } from "../utils";
+
+describe("actions", () => {
+  it("records state", async () => {
+    const { store } = model.createStore({
+      initState,
+    });
+
+    const { state } = await store.dispatch(recordState)(testAppState);
+    expect(Object.keys(state.app.state)).toHaveLength(
+      Object.keys(testAppState).length,
+    );
+
+    const recordedState = state.app.state;
+    expect(recordedState.foo).toBe("bar");
+    expect(recordedState.items[0]).toBe(1);
+    expect(recordedState.test.a).toBe("b");
```

Comment:
Instead of testing each element of the state individually, you can also just test the entire state with `toEqual`. And you could actually probably just check the state in the devtools universe completly equals `testAppState`

```ts
expect(recordedState).toEqual(testAppState)
```